packages/cli/internal/pkg/cli/spec/context.go

```diff
@@ -1,19 +1,46 @@
 package spec
 
+import "fmt"
+
 const DefaultMaxVCpus = 256
+const DefaultFSProvisionedThroughput = 0
 
+type FSConfig struct {
+	FSProvisionedThroughput int `yaml:"provisionedThroughput"`
+}
+type Filesystem struct {
+	FSType        string   `yaml:"fsType"`
+	Configuration FSConfig `yaml:"configuration,omitempty"`
+}
 type Engine struct {
-	Type   string `yaml:"type"`
-	Engine string `yaml:"engine"`
+	Type       string     `yaml:"type"`
+	Engine     string     `yaml:"engine"`
+	Filesystem Filesystem `yaml:"filesystem,omitempty"`
 }
-
 type Context struct {
 	InstanceTypes        []string `yaml:"instanceTypes,omitempty"`
 	RequestSpotInstances bool     `yaml:"requestSpotInstances,omitempty"`
 	MaxVCpus             int      `yaml:"maxVCpus,omitempty"`
 	Engines              []Engine `yaml:"engines"`
 }
 
+func (filesystem *Filesystem) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	type defValFilesystem Filesystem
```

Comment:
Why are you using a type alias here? You should be able to just use `Filesystem{Configuration:...}`